provd/go.mod

```diff
@@ -1,14 +1,14 @@
 module github.com/canonical/ubuntu-desktop-provision/provd
 
-go 1.21.3
+go 1.21.4
```

Comment:
is that needed? Are we vulnerable to some security issue fixed in `.4`?

Reply:
Yes, they were caught when I ran the sanity check the first time. Dependabot also catches them [here](https://github.com/canonical/ubuntu-desktop-provision/security/dependabot)